x-pack/qa/reindex-tests-with-security/build.gradle

```diff
@@ -31,13 +57,46 @@ integTestCluster {
                  'bin/elasticsearch-users', 'useradd', user, '-p', 'x-pack-test-password', '-r', role
   }
   waitCondition = { node, ant ->
-    File tmpFile = new File(node.cwd, 'wait.success')
-    ant.get(src: "http://${node.httpUri()}/_cluster/health?wait_for_nodes=>=${numNodes}&wait_for_status=yellow",
-            dest: tmpFile.toString(),
-            username: 'test_admin',
-            password: 'x-pack-test-password',
-            ignoreerrors: true,
-            retries: 10)
-    return tmpFile.exists()
+    // Load the CA PKCS#12 file as a truststore
+    KeyStore ks = KeyStore.getInstance("PKCS12");
+    ks.load(caFile.newInputStream(), 'password'.toCharArray());
+    TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+    tmf.init(ks);
+
+    // Configre a SSL context for TLS1.2 using our CA trust manager
+    SSLContext sslContext = SSLContext.getInstance("TLSv1.2");
+    sslContext.init(new KeyManager[0], tmf.getTrustManagers(), new SecureRandom());
+
+    // Check whether the cluster has started
+    URL url = new URL("https://${node.httpUri()}/_cluster/health?wait_for_nodes=${numNodes}&wait_for_status=yellow");
+    for (int i = 0; i < 20; i++) {
+      // we use custom wait logic here for HTTPS
```

Comment:
I think it would be awesome to unify this wait logic with that in `smoke-test-plugins-ssl`, potentially as a followup. Maybe @atorok has some guidance on the best way to do that?

Reply:
Yes, I think we probably want to replace the existing Ant task we use for this with our own task that can read from a custom truststore (or perhaps PEM so it can run on a FIPS JVM).